org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandler.java

```diff
@@ -311,4 +312,39 @@ private IRegion getRegion(ICompilationUnit cu, IDocument document, Position posi
 		return null;
 	}
 
+	/**
+	 * A utility function to format a String with given formatter options.
+	 *
+	 * @see org.eclipse.jdt.core.formatter.CodeFormatterApplication#formatFile(File,
+	 *      CodeFormatter)
+	 * @param content
+	 *                    the content to format
+	 * @param options
+	 *                    the Map includes formatter options
+	 * @param monitor
+	 *                    the progress monitor
+	 *
+	 * @return the formatted String
+	 */
+	public String stringFormatting(String content, Map<String, String> options, IProgressMonitor monitor) {
+		IDocument document = new Document();
+		document.set(content);
+		CodeFormatter formatter = ToolFactory.createCodeFormatter(options);
+		IRegion region = new Region(0, document.getLength());
+		TextEdit edit = formatter.format(CodeFormatter.K_COMPILATION_UNIT | CodeFormatter.F_INCLUDE_COMMENTS, content, region.getOffset(), region.getLength(), 0, TextUtilities.getDefaultLineDelimiter(document));

```

Comment:
You should follow the preference to determine whether to format the comments as well.

https://github.com/eclipse/eclipse.jdt.ls/blob/387304cdd5a440b3ab088ad2b9b4a52de8110848/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandler.java#L99

Reply:
I think it may confuse users. If the user set `java.format.comments.enabled` to `false` in VS Code settings, when he opens the Java Formatter Setting page, he has no way to preview the comment formatting again, since there is no setting controlling **whether** to format comments in the setting profile.

An optional way is to offer a new setting in the setting page to control whether to format comments. The value of this setting can be bundled with `java.format.comments.enabled`.

Reply:
To keep the behavior consistent with real java code formatting, we'll apply `preferenceManager.getPreferences().isJavaFormatComments()`